src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,40 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
+        if (isCorrectUser(user)) {
+            storageDao.add(user);
+            return user;
+        }
         return null;
     }
+
+    private boolean isCorrectUser(User user) {
+        if (user == null) {
+            throw new NullPointerException("User cannot be null!");
```

Comment:
lets throw ValidationException
NullPointerException should never be thrown from code